src/mito2/src/read/unordered_scan.rs

```diff
@@ -253,14 +253,14 @@ async fn maybe_init_parts(
 /// is no output ordering guarantee of each partition.
 #[derive(Default)]
 struct UnorderedDistributor {
-    file_ranges: Vec<FileRange>,
+    file_ranges: Vec<Box<dyn FileRange>>,
```

Comment:
Making the file range a trait may not be a good idea. We may create many trait objects because we will create a file range for each row group.